### DISCUSSION TITLE
print_orphan_nodes() monitoring

### DISCUSSION THREAD

**Discussion by Dustin00** [Ark2000/PankuConsole#145] [Ideas]
It'd be cool if I could turn on "Orphan Monitoring" with a set interval and then whenever print_orphan_nodes() actually sees something, Panku logged or popped up an alert that I have a leak.

When I realized print_orphan_nodes() was a thing, I immediately found a couple ways I was leaking, hunted them down, and ended them.

Now every 20 seconds my app is calling this so if I ever start leaking orphans again, I can address it immediately.

**Comment by Ark2000** [comment]
I like the idea that the plugin will be able detect potential memory leaking caused by orphan nodes, also

- we can use `Performance.get_monitor(Performance.OBJECT_ORPHAN_NODE_COUNT)` to get the orphan nodes count directly.

- sometimes orphan nodes are intentional thus there should be a count threshold to trigger the warning.

I highly recommend opening a feature proposal in the issues section. This will allow for a more detailed discussion and collaboration with the community.

**Reply by Dustin00** [reply]
Proposal opened: https://github.com/Ark2000/PankuConsole/issues/146

As I get better at Godot, hopefully I can help make contributions and not just pester you/the project.
